include/zephyr/bluetooth/mesh/access.h

```diff
@@ -1118,6 +1118,48 @@ bool bt_mesh_model_is_extended(struct bt_mesh_model *model);
  */
 int bt_mesh_comp_change_prepare(void);
 
+/** Composition data page 2 record. */
+struct bt_mesh_comp2_record {
+	/** Mesh profile ID. */
+	uint16_t id;
+	/** Version. */
+	struct {
+		/** Major version. */
+		uint8_t x;
+		/** Minor version. */
+		uint8_t y;
+		/** Z version. */
+		uint8_t z;
+	} version;
+	/** Element offset count. */
+	uint8_t elem_offset_cnt;
+	/** Element offset list. */
+	uint8_t *elem_offset;
+	/** Length of additional data. */
+	uint16_t data_len;
+	/** Additional data. */
+	void *data;
+};
+
+/** Node Composition data page 2 */
+struct bt_mesh_comp2 {
+	/** The number of NLC record on this device. */
+	size_t record_cnt;
+	/** List of records. */
+	struct bt_mesh_comp2_record *rec;
+};
+
+/** @brief Register composition data page 2 of the device.
+ *
+ *  Register Network Lighting Control (NLC) information for composition data
```

Comment:
This is the only place I see NLC mentioned, whereas the rest of this PR looks completely generic. Is this intentional?

Reply:
It is what the latest _published_ mesh spec assumes by `Composition Data Page 2 shall be present on a node if the node supports one or more mesh profile specifications [34].` (section 4.2.2.3 of MshPRFd1.1.r20). And it is this thing: https://www.bluetooth.com/learn-about-bluetooth/use-cases/lighting-control/. NLC drafts are published here: https://www.bluetooth.com/specifications/specs/?types=specs-docs&keyword=mesh Assigned numbers document calls them `Mesh Profiles` in section 3.12 (https://www.bluetooth.com/specifications/assigned-numbers/).

We need add documentation page for CDP2 in this PR. And I think in the API we should use `Mesh Profiles` rather than `NLC` with the reference to the Assigned numbers document section 3.12.

Reply:
I see your point @jhedberg. The reason why I mentioned it here is that I think the reference to comp data page # is a little abstract at times, and that it would be nice to have at least one mention of what comp page 2 actually contains. But I see that it might be a little inconsistent.

Reply:
Seem like the docs for composition data pages is somewhat lacking. My best guess is to put it as a separate section under Bluetooth Mesh Profile->Access layer in the docs tree? I think I will cover all of them (0,1,2,128,129,130) for consistency.